scintilla/src/Editor.cxx

```diff
@@ -4361,14 +4363,16 @@ void Editor::CopySelectionRange(SelectionText *ss, bool allowLineCopy) {
 		std::vector<SelectionRange> rangesInOrder = sel.RangesCopy();
 		if (sel.selType == Selection::SelTypes::rectangle)
 			std::sort(rangesInOrder.begin(), rangesInOrder.end());
-		for (const SelectionRange &current : rangesInOrder) {
-			text.append(RangeText(current.Start().Position(), current.End().Position()));
-			if (rangesInOrder.size() > 1) {
-				text.append(pdoc->EOLString());
+		const std::string_view separator = (sel.selType == Selection::SelTypes::rectangle) ? pdoc->EOLString() : copySeparator;
+		for (size_t part = 0; part < rangesInOrder.size(); part++) {
+			text.append(RangeText(rangesInOrder[part].Start().Position(), rangesInOrder[part].End().Position()));
+			if ((sel.selType == Selection::SelTypes::rectangle) || (part < rangesInOrder.size() - 1)) {
+				// Append unless simple selection or last part of multiple selection
+				text.append(separator);
 			}
 		}
 		ss->Copy(text, pdoc->dbcsCodePage,
-			vs.styles[StyleDefault].characterSet, rangesInOrder.size() > 1, sel.selType == Selection::SelTypes::lines);

```

Comment:
@donho Conflict between 4ff9d77d2bfbb323e2fcba10f02570710872db94 and the current modification by scintilla 5.5.2 update which still needs to be checked and adapted.

Reply:
@chcg 
IMO, for not having regression, `vs.styles[StyleDefault].characterSet, sel.IsRectangular(), sel.selType == Selection::SelTypes::lines);` should be `vs.styles[StyleDefault].characterSet, rangesInOrder.size() > 1, sel.selType == Selection::SelTypes::lines);`.
I'll test it and let you know the test result.

Reply:
@chcg 
After checking the old behaviour, the above modification is necessary, but not enough.
We should also change from:
`if ((sel.selType == Selection::SelTypes::rectangle) || (part < rangesInOrder.size() - 1)) {`
to
`if ((rangesInOrder.size() > 1) || (part < rangesInOrder.size() - 1)) {`

and as well from:
`const std::string_view separator = (sel.selType == Selection::SelTypes::rectangle) ? pdoc->EOLString() : copySeparator;`
to
`const std::string_view separator = (rangesInOrder.size() > 1) ? pdoc->EOLString() : copySeparator;`